### PR TITLE
chore(site): add tests for deprecate template flow

### LINKED ISSUE #12505: Test deprecate template flow
Part of this issue is determining whether an e2e test is necessary for listing templates. We do not support querying for this page so it may be suitable to simply have a unit test here.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Closes #12505

### CODE REVIEW

File: site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx [hunk 1]
```diff
@@ -126,4 +131,66 @@ describe("TemplateSettingsPage", () => {
...
+        }),
+      );
+      const updateTemplateMetaSpy = jest.spyOn(API, "updateTemplateMeta");
+      const deprecationMessage = "This template is deprecated";
+
+      await renderTemplateSettingsPage();
+      await deprecateTemplate(MockTemplate, deprecationMessage);
+
+      const [templateId, data] = updateTemplateMetaSpy.mock.calls[0];
+      await waitFor(() => expect(templateId).toEqual(MockTemplate.id));
```
Comment (code-asher) [inline site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx:154]: Does this `waitFor` do anything?  It looks like it is comparing two static IDs so the result would never change, or am I missing something?

File: site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx [hunk 2]
```diff
@@ -126,4 +131,66 @@ describe("TemplateSettingsPage", () => {
...
+      const updateTemplateMetaSpy = jest.spyOn(API, "updateTemplateMeta");
+
+      await renderTemplateSettingsPage();
+      await deprecateTemplate(
+        MockTemplate,
+        "This template should not be able to deprecate",
+      );
+
+      const [templateId, data] = updateTemplateMetaSpy.mock.calls[0];
+      await waitFor(() => expect(templateId).toEqual(MockTemplate.id));
```
Comment (code-asher) [inline site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx:180]: Same question here about `waitFor`.

**Review by code-asher — APPROVED** [review]
Looks good!

File: site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx:154]: You are right

**Comment by BrunoQuaresma** [comment]
### Merge activity

* **Mar 21, 9:37 AM EDT**: @BrunoQuaresma merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coder/coder/12685).

**Pull request merged**